google/resource_dataproc_cluster_test.go

```diff
@@ -0,0 +1,780 @@
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "init_bucket" {
+    name          = "%s"
+    force_destroy = "true"
+}
+
+resource "google_storage_bucket_object" "init_script" {
+  name           = "dproc-cluster-test-%s-init-script.sh"
+  bucket         = "${google_storage_bucket.init_bucket.name}"
+  content        = <<EOL
+#!/bin/bash
+ROLE=$$(/usr/share/google/get_metadata_value attributes/dataproc-role)
+if [[ "$${ROLE}" == 'Master' ]]; then
+  echo "on the master" >> /tmp/%s
+  gsutil cp /tmp/%s ${google_storage_bucket.init_bucket.url}
+else
+  echo "on the worker" >> /tmp/msg.txt
+fi
+EOL
+
+}
+
+resource "google_dataproc_cluster" "with_init_action" {
+	name   = "dproc-cluster-test-%s"
+	region = "us-central1"
+	initialization_action_timeout_sec = 500
+	initialization_actions = [
+	   "${google_storage_bucket.init_bucket.url}/${google_storage_bucket_object.init_script.name}"
+	]
+
+    # Keep the costs down with smallest config we can get away with
+    properties = {
+        "dataproc:dataproc.allow.zero.workers" = "true"
+    }
+    worker_config { }
+	master_config {
+		machine_type      = "n1-standard-1"
+		boot_disk_size_gb = 10
+	}
+}`, bucket, rnd, objName, objName, rnd)
+}
+
+func testAccDataprocCluster_withStagingBucketOnly(bucketName string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+    name          = "%s"
+    force_destroy = "true"
+}`, bucketName)
+}
+
+func testAccDataprocCluster_withStagingBucketAndCluster(clusterName, bucketName string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "google_dataproc_cluster" "with_bucket" {
+	name   = "%s"
+	region = "us-central1"
+	staging_bucket = "${google_storage_bucket.bucket.name}"
+
+    # Keep the costs down with smallest config we can get away with
+    properties = {
+        "dataproc:dataproc.allow.zero.workers" = "true"
+    }
+    worker_config { }
+	master_config {
+		machine_type      = "n1-standard-1"
+		boot_disk_size_gb = 10
+	}
+}`, testAccDataprocCluster_withStagingBucketOnly(bucketName), clusterName)
+}
+
+func testAccDataprocCluster_withImageVersion(rnd string) string {
+	return fmt.Sprintf(`
+resource "google_dataproc_cluster" "with_image_version" {
+	name   = "dproc-cluster-test-%s"
+	region = "us-central1"
+	image_version = "preview"
+}`, rnd)
+}
+
+func testAccDataprocCluster_withServiceAcc(saEmail string, rnd string) string {
+	return fmt.Sprintf(`
+resource "google_dataproc_cluster" "with_service_account" {
+	name   = "dproc-cluster-test-%s"
+	region = "us-central1"
+
+    # Keep the costs down with smallest config we can get away with
+    properties = {
+        "dataproc:dataproc.allow.zero.workers" = "true"
+    }
+    worker_config { }
+	master_config {
+		machine_type      = "n1-standard-1"
+		boot_disk_size_gb = 10
+	}
+
+	service_account = "%s"
+
+	service_account_scopes = [
+        #    The following scopes necessary for the cluster to function properly are
```

Comment:
Can you fix the indentation on these comments? (some use tabs and some use spaces)